pkg/resourcemanager/psql/psqluser/psqluser.go

```diff
@@ -0,0 +1,285 @@
+
+	// TODO: Have db lib do string interpolation
+	//tsql := fmt.Sprintf(`CREATE USER @User WITH PASSWORD='@Password'`)
+	//_, err := db.ExecContext(ctx, tsql, sql.Named("User", newUser), sql.Named("Password", newPassword))
+
+	if err != nil {
+		return newUser, err
+	}
+	return newUser, nil
+}
+
+// UpdateUser - Updates user password
+func (s *PostgreSqlUserManager) UpdateUser(ctx context.Context, secret map[string][]byte, db *sql.DB) error {
+	user := string(secret[PSecretUsernameKey])
+	newPassword := helpers.NewPassword()
+
+	// make an effort to prevent sql injection
+	if err := findBadChars(user); err != nil {
+		return fmt.Errorf("Problem found with username: %v", err)
+	}
+	if err := findBadChars(newPassword); err != nil {
+		return fmt.Errorf("Problem found with password: %v", err)
+	}
+
+	tsql := fmt.Sprintf("ALTER USER '%s' WITH PASSWORD '%s'", user, newPassword)
+	_, err := db.ExecContext(ctx, tsql)
+
+	return err
+}
+
+// UserExists checks if db contains user
+func (s *PostgreSqlUserManager) UserExists(ctx context.Context, db *sql.DB, username string) (bool, error) {
+
+	tsql := fmt.Sprintf("SELECT * FROM pg_user WHERE usename = '%s'", username)
+	res, err := db.ExecContext(ctx, tsql)
+	if err != nil {
+		return false, err
+	}
+	rows, err := res.RowsAffected()
+	return rows > 0, err
+
+}
+
+// DropUser drops a user from db
+func (s *PostgreSqlUserManager) DropUser(ctx context.Context, db *sql.DB, user string) error {
+	tsql := fmt.Sprintf("DROP USER IF EXISTS %q", user)
+	_, err := db.ExecContext(ctx, tsql)
+	return err
+}
+
+// DeleteSecrets deletes the secrets associated with a SQLUser
+func (s *PostgreSqlUserManager) DeleteSecrets(ctx context.Context, instance *v1alpha1.PostgreSQLUser, secretClient secrets.SecretClient) (bool, error) {
+	// determine our key namespace - if we're persisting to kube, we should use the actual instance namespace.
+	// In keyvault we have some creative freedom to allow more flexibility
+	secretKey := GetNamespacedName(instance, secretClient)
+
+	// delete standard user secret
+	err := secretClient.Delete(
+		ctx,
+		secretKey,
+	)
+	if err != nil {
+		instance.Status.Message = "failed to delete secret, err: " + err.Error()
+		return false, err
+	}
+
+	// delete all the custom formatted secrets if keyvault is in use
+	keyVaultEnabled := reflect.TypeOf(secretClient).Elem().Name() == "KeyvaultSecretClient"
+	if keyVaultEnabled {
+		customFormatNames := []string{
+			"adonet",
+			"adonet-urlonly",
+			"jdbc",
+			"jdbc-urlonly",
+			"odbc",
+			"odbc-urlonly",
+			"server",
+			"database",
+			"username",
+			"password",
+		}
+
+		for _, formatName := range customFormatNames {
+			key := types.NamespacedName{Namespace: secretKey.Namespace, Name: instance.Name + "-" + formatName}
+
+			err = secretClient.Delete(
+				ctx,
+				key,
+			)
+			if err != nil {
+				instance.Status.Message = "failed to delete secret, err: " + err.Error()
+				return false, err
+			}
+		}
+	}
+
```

Comment:
I only left this comment to confirm with you since the code change is big.  if we remove the custom secret formating, so we would  not allow user to use keyVaultSecretPrefix but we still allow customer to use keyvault to store the secret, right?